pyiron_atomistics/lammps/structure.py

```diff
@@ -115,7 +115,7 @@ def fold(vec, pvec, i):
 
         self.A = apre
 
-        if self.is_skewed() and (not (pbc[0] and pbc[1] and pbc[2])):
+        if self._force_skewed and (not (pbc[0] and pbc[1] and pbc[2])):
```

Comment:
Is it by design that `self.is_skewed() == True` no longer triggers this check? I would perhaps assume that this could be the case, since simulations with skewed and non-PBC are safe as long as the pressure tensor does not touch the shear directions? This tripped me a bit, so maybe a comment explaining the rationale is in order?

Reply:
So your suggestion really totally makes sense, but `_force_skewed` is not part of the class and therefore we'd have to include it either as an argument or an attribute of this class, both of which would be somewhat ugly. Besides, since it's not a clear error, as long as the user knows what they are doing, I think it goes too far if we clearly forbid it. Therefore I replaced the error by a warning.

In any case, we're planning to overhaul the input structure in the near future, which would allow us to have pressures in the input as well, so if the solution I propose here somewhat solves a practical problem, I'd say we can merge it for now. Later when the input structure is there, we can make it more elegant.

Reply:
Erm, I'm not sure I understand your answer?

_You_ added the check using `self._force_skewed`, it was `self.is_skewed()` before. I was wondering if that was relaxing the checks too much, because you can now force skewed boxes in cases where it was not allowed before, namely when you do have a non-orthogonal box and no PBCs, but do not apply any shear stress. This behavior, introduced by your patch, might be OK or not.

Reply:
> You added the check using self._force_skewed

Yes, I added the check according to what you wrote, but then as the tests failed I realised that `_force_skewed` was not part of the class. I don't really see a nice way to check whether the PBC setting and skewness are compatible from the pyiron side, so I'd rather leave it to LAMMPS to decide.

Actually I don't really understand your comment either...

Reply:
To be honest, as long as we don't have a clear way to check whether the box setting is compatible with what LAMMPS allows, we should leave it to LAMMPS to decide whether it's allowed or not. Therefore I think it's totally appropriate to make it a warning and not an error here.

Reply:
I was about to write this:

> OK, let's make this easier: Why not emit a warning in `LammpsStructure.simulation_cell()`, which is the only place where the structure is written and which has both `is_skewed() and `_force_skewed` available?
> 
> Add the check inside the `if` here: https://github.com/pyiron/pyiron_atomistics/blob/ed3e79c2e599fbb7c302b5203167d2082ea61cd5/pyiron_atomistics/lammps/structure.py#L286

But then I realized that the `pbc` argument is not passed to `UnfoldingPrism.__init__()` here: https://github.com/pyiron/pyiron_atomistics/blob/ed3e79c2e599fbb7c302b5203167d2082ea61cd5/pyiron_atomistics/lammps/structure.py#L276

So I don't understand if this check in the current code is applied at all or if it completely useless. If you want to figure this out, go ahead, but I'm OK with the code as-is. Especially if an overhaul is planned.

I also agree that a warning in lieu of an error is appropriate.